lib/moesif_aws_lambda/moesif_aws_middleware.rb

```diff
@@ -62,13 +62,59 @@ def base64_encode_body(body)
       return Base64.encode64(body), "base64"
     end
 
+    def is_base64_str(body)
+      if not body.instance_of?(String)
+        return false
+      end
+      if body.length % 4 != 0
+        return false
+      end
+
+      b64_regex = /^[A-Za-z0-9+\/]+={0,2}$/
+
+      if not body.match?(b64_regex)
+        return false
+      end
+
+      begin
+        _ = Base64.decode64(body)
+        return true
+      rescue
+        return false
+      end
+
+    end
+
     def @moesif_helpers.log_debug(message)
       if @debug
         puts("#{Time.now.to_s} [Moesif Middleware] PID #{Process.pid} TID #{Thread.current.object_id} #{message}")
       end
     end
 
-    def parse_body(body, headers)
+    def process_body(body_wrapper, headers)
+      # the `event` object can either be a Hash or a String
+      if body_wrapper.instance_of?(Hash)
+        if body_wrapper.key?("body")
+          begin
+            if body_wrapper.fetch("isBase64Encoded", false) and is_base64_str(body_wrapper.fetch("body"))
```

Comment:
since we're using `body_wrapper.fetch("body")` multiple times/places, we'd just fetch it once and store it in variable body = body_wrapper.fetch("body") and then use that variable everywhere